alarms_agent.cpp

```diff
@@ -51,38 +54,76 @@ bool done = false;
 // Signal handler that triggers termination.
 void terminate_handler(int sig)
 {
+  snmp_terminate("clearwater-alarms");
   done = true;
 }
 
+enum OptionTypes
+{
+  OPT_COMMUNITY=256+1,
+  OPT_SNMP_IPS,
+  OPT_LOG_LEVEL,
+  OPT_LOG_DIR
+};
+
+const static struct option long_opt[] =
+{
+  { "community",                       required_argument, 0, OPT_COMMUNITY},
+  { "snmp-ips",                        required_argument, 0, OPT_SNMP_IPS},
+  { "log-level",                       required_argument, 0, OPT_LOG_LEVEL},
+  { "log-dir",                         required_argument, 0, OPT_LOG_DIR},
+};
+
+static void usage(void)
+{
+    puts("Options:\n"
+         "\n"
+         " --snmp-ips <ip>,<ip>       Send SNMP notifications to the specified IPs\n"
+         " --community <name>         Include the given community string on notifications\n"
+         " --log-dir <directory>\n"
+         "                            Log to file in specified directory\n"
+         " --log-level N              Set log level to N (default: 4)\n"
+        );
+}
+
 int main (int argc, char **argv)
 {
   std::vector<std::string> trap_ips;
   char* community = NULL;
+  std::string logdir = "";
+  int loglevel = 4;
   int c;
+  int optind;
 
+  printf("Helo world\n");
   opterr = 0;
-  while ((c = getopt (argc, argv, "i:c:")) != -1)
+  while ((c = getopt_long(argc, argv, "", long_opt, &optind)) != -1)
   {
     switch (c)
       {
-      case 'c':
+      case OPT_COMMUNITY:
         community = optarg;
         break;
-      case 'i':
+      case OPT_SNMP_IPS:
         Utils::split_string(optarg, ',', trap_ips);
         break;
+      case OPT_LOG_LEVEL:
+        loglevel = atoi(optarg);
+        break;
+      case OPT_LOG_DIR:
+        logdir = optarg;
+        break;
       default:
-        abort ();
+        usage();
+        abort();
       }
   }
-
-  // Log SNMP library output to syslog
-  snmp_enable_calllog();
-  snmp_enable_syslog_ident("alarms_agent", LOG_DAEMON);
-
-  // Set ourselves up as a subagent
-  netsnmp_ds_set_boolean(NETSNMP_DS_APPLICATION_ID, NETSNMP_DS_AGENT_ROLE, 1);
-  init_agent("clearwater-alarms");
+  
+  Log::setLoggingLevel(loglevel);
+  Log::setLogger(new Logger(logdir, "clearwater-alarms"));
+  TRC_ERROR("I have started");
```

Comment:
I agree that it's odd that this is an error, but I'm a big fan of writing logs in the first person. This could help us engage better with debugging problems.